core/sql/executor/ExExeUtilCommon.cpp

```diff
@@ -669,7 +669,9 @@ short ExExeUtilTcb::executeQuery(char * task,
 	    char * stringParam1 = NULL;
 	    Lng32   intParam1 = ComDiags_UnInitialized_Int;
 
-	    retcode = (short)cliInterface()->retrieveSQLDiagnostics(getDiagsArea());
+            setDiagsArea(cliInterface()->allocAndRetrieveSQLDiagnostics(getDiagsArea()));
+            if (getDiagsArea() != NULL)
+	        retcode = 0;
```

Comment:
Should we rely on the assumption that retcode is non-zero when we reach here? Maybe better to set it explicitly to some defined value, something indicating that an error occurred but that we found no diagnostics information?

Reply:
Yes. We can rely on that assumption. It is strange that the earlier code sets the sqlcode to the return code of SQL_EXEC_MergeDiagnosticsInternal to be the sqlcode.  See line 700 below. When we are able to obtain the diagnostics area, the sqlcode is set the main error code

Reply:
Thanks for the explanation.